src/pudl/dagster_etl.py

```diff
@@ -0,0 +1,81 @@
+"""Dagster version of EPA CEMS ETL."""
+from pathlib import Path
+
+import pandas as pd
+import sqlalchemy as sa
+from dagster import (AssetMaterialization, EventMetadata, Field, In, Output,
+                     job, op, resource)
+
+import pudl
+from pudl.helpers import compute_dataframe_summary_statistics
+from pudl.metadata.classes import Resource
+from pudl.workspace.datastore import Datastore
+
+# TODO: how can we associate metadata schemas with specific ops?
+EpaCemsDataFrame = Resource.from_id("hourly_emissions_epacems").to_dagster(
+    event_metadata_fn=compute_dataframe_summary_statistics)
+
+
+@op(required_resource_keys={"pudl_settings"}, ins={"transformed_df": In(dagster_type=EpaCemsDataFrame)})
+def load_epacems(context, transformed_df: pd.DataFrame):
+    """Load epacems to parquet."""
+    root_path = Path(context.resources.pudl_settings["parquet_dir"]) / "epacems"
+
+    pudl.load.df_to_parquet(
+        transformed_df,
+        resource_id="hourly_emissions_epacems",
+        root_path=root_path,
+        partition_cols=["year", "state"]
+    )
+
+    # Format of the mapping key makes it difficult to recreate the final parquet path.
+    yield AssetMaterialization(
```

Comment:
I'm not entirely sure how Dagster handles yields. It is strange to yield multiple objects from a function. There is a way to [link assets to outputs](https://docs.dagster.io/concepts/assets/asset-materializations#linking-assets-to-an-output-definition) so you don't have to yield an AssetMaterialization and an Output.

Reply:
You can have as many yields as you want -- in our current cems code we have 1250 yields in the extract function:
```
for year in epacems_years:
    for state in states:
        ...
        yield df
```        
It's like returning a list or tuple, but where each item is computed (and stored) as-needed instead of all at once.
I think Dagster has to use generators to avoid running out of memory on big parallel jobs like CEMS. They don't know how big your objects are in advance, and if you're making 1250 of them they don't have to be that big to blow up the process.

Reply:
I'm used to seeing in the context of a lazy list (generator) like the epacems, but in that case all of the things that are being `yield`ed are the same *kind* of thing -- I think what feels a little weird here is that there are potentially several different types of things being `yield`ed sequentially, which implies there's a matching listener on the other end expecting to hear a particular sequence of things... or I guess it's probably something like an [Observer pattern](https://en.wikipedia.org/wiki/Observer_pattern) where it's waiting to get a few different kinds of notifications? And then choosing the appropriate action to take based on what it saw/heard.